tests/test_webservice.py

```diff
@@ -25,8 +23,21 @@
 class BaseTest(unittest.TestCase):
     client_class: Union[Type[AsyncClient], Type[Client]] = Client
 
+    @pytest.fixture(autouse=True)
+    def setup_httpserver(self, httpserver: HTTPServer):
+        self.httpserver = httpserver
+
     def setUp(self):
         self.client = self.client_class(42, "abcdef123456")
+        self.client._base_uri = self.httpserver.url_for("/minfraud/v2.0")
+        self.client._factors_uri = self.httpserver.url_for("/minfraud/v2.0/factors")
+
+        self.client._insights_uri = self.httpserver.url_for("/minfraud/v2.0/insights")
+        self.client._score_uri = self.httpserver.url_for("/minfraud/v2.0/score")
+        self.client._report_uri = self.httpserver.url_for(
+            "/minfraud/v2.0/transactions/report"
+        )
```

Comment:
We shouldn't be modifying _many_ private attributes on the client class. I probably should have raised this on the `geoip2` one, but there it was just one. Beyond modifying the internal state of the object in a confusing way, this bypasses any testing of the constructor.

I think it would be better to do something more similar to [this](https://github.com/maxmind/minfraud-api-python/commit/7ee97378ecf38505e7892b23012912cb20d5d195#diff-e9ad8437a072a618b0ee22fd49bf2461273e4a6977ac87b29478175b4c03523bL37) and then passing the host and port through using the `host` paramter.

Reply:
how do you want the scheme (i.e. http or https) to be set for the client?  pytest_httpserver runs on http, to run it as https we need another dependancy as described in this doc: https://pytest-httpserver.readthedocs.io/en/latest/howto.html#running-an-https-server

Reply:
See the link in my comment for how we set the scheme temporarily when there was an issue with `mocket`. I think we can just do that again. I don't think we want to make it part of the public API.